system/doc/reference_manual/ref_man_records.md

```diff
@@ -55,6 +54,29 @@ definition is placed in an include file.
 > using the syntax described in this section. In earlier releases, it was
 > necessary to use the `m:shell` built-in function `rd/2`.
 
+> #### Change {: .info }
+>
+> In Erlang/OTP 28, variables are allowed in the record definition.
+> If a record definition contains a variable, a function call will be made to
+> initialize the records omitted fields.
+
+```erlang
+-record(r, {a = case my_f() of {X, _} -> X; _ -> ok end,
+            b = case my_g() of {Y, _} -> Y; _ -> ok end}).
+t(X) ->
+   #r{}. %% X will not have any effect on initialization of #r
+```
+
+Variables bound in a field are visible in the fields defined after
+so use with caution.
+```erlang
+-record(r, {a = case my_f() of {Res, _} -> Res; _ -> ok end,
+            b = case my_g() of {Res, _} -> Res; _ -> error end}).
+```
+Variables need to be bounded, which is not the case in the following.
```

Comment:
Are you trying to say that the following example will cause a compiler error? If so, I suggest that you show the compiler error.

```suggestion
Variables need to be bound, which is not the case in the following.
```